qwandaq/src/main/java/life/genny/qwandaq/entity/search/SearchEntity.java

```diff
@@ -653,7 +653,10 @@ public <T extends Trait> Optional<T> getTrait(Class<T> trait, String traitCode)
 	@JsonbTransient
 	public <T extends Trait> List<T> getTraits(Class<T> traitType) {
 		List<T> t = traits.get(traitType);
-		if(t == null) return new ArrayList<>();
+		if(t == null){
+			t = new ArrayList<>();
+			traits.put(traitType, t);
+		}
```

Comment:
no point just returning a new arraylist if its not going to get linked up to something